source/bsd/kqueue_event_loop.c

```diff
@@ -711,8 +712,8 @@ static int s_unsubscribe_from_io_events(struct aws_event_loop *event_loop, struc
 static bool s_is_event_thread(struct aws_event_loop *event_loop) {
     struct kqueue_loop *impl = event_loop->impl_data;
 
-    uint64_t thread_id = aws_atomic_load_int(&impl->thread_id);
-    return aws_thread_current_thread_id() == thread_id;
+    aws_thread_id *thread_id = aws_atomic_load_ptr(&impl->thread_id);
+    return thread_id && aws_thread_thread_id_equal(*thread_id, aws_thread_current_thread_id());
```

Comment:
What is the semantics of a `null` pointer here.  Should this be an assert?

Reply:
`NULL` is ok. For example:

```c
// in client:
struct aws_event_loop *el = aws_event_loop_new_default(...); //< ptr set to NULL
bool b = aws_event_loop_thread_is_callers_thread(el); //< okay to call this even though event loop is not running so will return false
```